flink-clients/src/main/java/org/apache/flink/client/program/ClusterClient.java

```diff
@@ -45,37 +43,9 @@
  */
 public abstract class ClusterClient<T> implements AutoCloseable {
 
-	/** Configuration of the client. */
-	private final Configuration flinkConfig;
-
-	protected JobExecutionResult lastJobExecutionResult;
-
 	/** Switch for blocking/detached job submission of the client. */
 	private boolean detachedJobSubmission = false;
 
-	// ------------------------------------------------------------------------
-	//                            Construction
-	// ------------------------------------------------------------------------
-
-	/**
-	 * Creates a instance that submits the programs to the JobManager defined in the
-	 * configuration. This method will try to resolve the JobManager hostname and throw an exception
-	 * if that is not possible.
-	 *
-	 * @param flinkConfig The config used to obtain the job-manager's address, and used to configure the optimizer.
-	 */
-	public ClusterClient(Configuration flinkConfig) {
-		this.flinkConfig = Preconditions.checkNotNull(flinkConfig);
-	}
-
-	/**
-	 * User overridable hook to close the client, possibly closes internal services.
-	 * @deprecated use the {@link #close()} instead. This method stays for backwards compatibility.
-	 */
-	public void shutdown() throws Exception {
-		close();
```

Comment:
Hi @TisonKun, personally I am up for removing this method in favour of the `close()`, and in fact I had removed it in one of my old PRs and re-added it after @zentol said that people are using the `ClusterClient` directly. So I would say let's wait for @zentol to comment on that, but, as I said, I would also vote for removing it.